src/openrct2/object/RideObject.cpp

```diff
@@ -370,6 +374,53 @@ void RideObject::DrawPreview(rct_drawpixelinfo* dpi, [[maybe_unused]] int32_t wi
     gfx_draw_sprite(dpi, imageId, 0, 0, 0);
 }
 
+bool RideObject::ValidateObject() const
+{
+    int16_t max_height = std::numeric_limits<int16_t>::min();
+    int16_t max_width = std::numeric_limits<int16_t>::min();
+    for (uint32_t i = 0; i < GetImageTable().GetCount(); i++)
+    {
+        if (GetImageTable().GetImages()[i].height > max_height)
+        {
+            max_height = GetImageTable().GetImages()[i].height;
+        }
+        if (GetImageTable().GetImages()[i].width > max_width)
+        {
+            max_width = GetImageTable().GetImages()[i].width;
+        }
+    }
+    if (max_width < 0 || max_height < 0)
+    {
+        return false;
+    }
+    size_t numPixels = max_width * max_height;
+    auto pixels8 = new uint8_t[numPixels];
+    std::fill_n(pixels8, numPixels, 0);
+
+    rct_drawpixelinfo dpi;
+    dpi.bits = pixels8;
+    dpi.pitch = 0;
+    dpi.x = 0;
+    dpi.y = 0;
+    dpi.width = max_width;
+    dpi.height = max_height;
+    dpi.zoom_level = 0;
+
+    uint32_t imageId = _legacyType.images_offset;
```

Comment:
It was copied from https://github.com/OpenRCT2/OpenRCT2/blob/0c4623a39fa5dec8eac1c45f358e0fd45fd2f1df/src/openrct2/object/RideObject.cpp#L360-L368 and this is likely wrong/insufficient, would appreciate hints on how to get correct values.

Reply:
Looks fine to me. This will slip over all the preview images for the ride and get to the sprites